project_close_forecast_date/models/purchase_order.py

```diff
@@ -0,0 +1,15 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
+from pytz import timezone
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    @api.onchange('date_planned')
+    def _onchange_date_planned(self):
+        if self.order_line:
+            tz = timezone(self.env.user.tz or self.company_id.partner_id.tz or 'UTC')
+            self.order_line.forecast_date = self.date_planned.astimezone(tz).date()

```

Comment:
This process is repeated below, then should be moved to an order line method, in order to not be repeated. I suspect that there's an Odoo base method that makes such conversion, if I find it, I'll tell you